app/code/Magento/InventoryConfiguration/Model/ResourceModel/GetConfigurationValue.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfiguration\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * If $stockId && $sourceCode && $sku are NULL => GLOBAL CONFIG
+ * If specified $stockId only => STOCK CONFIG
+ * If specified $stockId && $sku => STOCK ITEM CONFIG
+ * If specified $sourceCode only => SOURCE CONFIG
+ * If specified $sourceCode && $sku => SOURCE ITEM CONFIG
+ */
+class GetConfigurationValue
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(ResourceConnection $resourceConnection)
+    {
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @param string $configOption
+     * @param string|null $sku
+     * @param string|null $sourceCode
+     * @param int|null $stockId
+     * @return null|string
+     */
+    public function execute(
+        string $configOption,
+        int $stockId = null,
+        string $sourceCode = null,
+        string $sku = null
+    ): ?string {
+        $connection = $this->resourceConnection->getConnection();
+        $inventoryConfigurationTable = $this->resourceConnection->getTableName('inventory_configuration');
+
+        $select = $connection->select()
+            ->from($inventoryConfigurationTable, 'value')
+            ->where('config_option = ?', $configOption)
+            ->limit(1);
+
+        if (isset($stockId)) {
```

Comment:
Hi @seruymt, I don't understand why, when `$stockId`, `$sourceCode`, and `$sku` are set you add the `IS NULL` clause.
I may be wrong but shouldn't it be the opposite? 
Or am I missing something? 
Thank you

Reply:
Hi @aleron75, thanks for your comment. This is a way how we divide config option on scopes (Store, StoreItem, Source, SourceItem and Global). You can read more about it here - https://github.com/magento-engcom/msi/wiki/Inventory-configuration-Design-and-DB-structure

Reply:
Hi @seruymt I've read the document you pointed out. There, for example, I read:

>To retrieve Source level configuration options we need to execute next query:
>`select * from inventory_configuration where source_code = %source_code% and sku is NULL and stock_id is NULL`

as far as I understand by the example above, to retrieve a value at source level we only use the following match: `source_code = %source_code%`

so I still miss the point in committed code when, for example at line 61 of `GetConfigurationValue` class, if `$sourceCode` is set, we generate a `$select->where('source_code IS NULL');` statement instead of a `$select->where('source_code = ?', $sourceCode);`

Thanks